conn.go

```diff
@@ -479,7 +479,11 @@ func (c *Conn) readLoop() {
 		}
 
 		frameType, data, err := ReadUnpackedResponse(c)
+
 		if err != nil {
+			if err == io.EOF && atomic.LoadInt32(&c.closeFlag) == 1 {
+				goto exit
+			}
 			if !strings.Contains(err.Error(), "use of closed network connection") {
 				c.log(LogLevelError, "IO error - %s", err)
 				c.delegate.OnIOError(c, err)
```

Comment:
My assumption is that there's an edge case where you're cleanly shutting down but you _don't_ "use a closed connection" but instead hit an "expected" EOF.

This person was clearly getting a log message despite the existing logic, which I think confirms that.